package.json

```diff
@@ -137,6 +138,8 @@
     "test-unit": "jest",
     "test-unit:coverage": "jest --coverage",
     "test-unit:coverage-ci": "jest --coverage --maxWorkers 1 && codecov",
-    "test-unit:watch": "jest --watch"
+    "test-unit:watch": "jest --watch",
+    "test-e2e": "cypress run",
```

Comment:
Should these be running `bin/run-e2e-tests.sh` ? When switching to this branch again and running `npm run test-e2e` it showed me a cryptic error because the Docker instance wasn't running:

```
Opening Cypress...
Cypress could not verify that the server set as your 'baseUrl' is running:

  > http://localhost:8888
```

Without context that `bin/run-e2e-tests.sh` even exists, I think this could be cause for confusion for new contributors.

Reply:
Good question! If we switch this to `bin/run-e2e-tests.sh` this means for development purpose, this is useless since you do not want to setup the environment on each run. 

What if we add a `test-e2e:ci` that runs  `bin/run-e2e-tests.sh`?

Reply:
Would the following setup work for you?:
```json
"docker:setup-if-not-exists": "./bin/setup-local-env.sh",
"pretest-e2e": "npm run docker:setup-if-not-exists",
"test-e2e": "cypress run",
"pretest-e2e:watch": "npm run docker:setup-if-not-exists",
"test-e2e:watch": "cypress open"
```

This assumes that `docker:setup-if-not-exists` checks if docker instance is already there. If not it setups everything. This way it would work with both CI and local development.

Reply:
Yeah it seems like we could use some `pre` hooks on these tasks that run the setup if it's not already available (idempotent setup).